mailbox/plugin/deleted-messages-vault/src/main/java/org/apache/james/vault/utils/DeleteByQueryExecutor.java

```diff
@@ -54,13 +57,15 @@
     private static final Logger LOGGER = LoggerFactory.getLogger(DeleteByQueryExecutor.class);
 
     private final DeletedMessageVault deletedMessageVault;
+    private final Supplier<Publisher<User>> userWithVaults;
 
-    public DeleteByQueryExecutor(DeletedMessageVault deletedMessageVault) {
+    public DeleteByQueryExecutor(DeletedMessageVault deletedMessageVault, Supplier<Publisher<User>> userWithVaults) {
```

Comment:
I think you can pass `Publisher<User> userWithVaults` instead. Publisher here can be a lazy supplier, so I think don't need that Supplier wrapper